addons/hr_holidays/security/hr_holidays_security.xml

```diff
@@ -93,36 +93,44 @@
     </record>
 
     <record id="hr_leave_allocation_rule_employee" model="ir.rule">
-        <field name="name">Allocations: employee: read own</field>
+        <field name="name">Allocations: employee: read all</field>
         <field name="model_id" ref="model_hr_leave_allocation"/>
-        <field name="domain_force">[
-            '|',
-                ('employee_id.leave_manager_id', '=', user.id),
-                ('employee_id.user_id', '=', user.id)
-        ]</field>
+        <field name="domain_force">[(1, '=', 1)]</field>
         <field name="perm_create" eval="False"/>
         <field name="perm_write" eval="False"/>
         <field name="perm_unlink" eval="False"/>
         <field name="groups" eval="[(4,ref('base.group_user'))]"/>
     </record>
 
     <record id="hr_leave_allocation_rule_employee_update" model="ir.rule">
-        <field name="name">Allocations: employee: create/write/unlink not validated own</field>
+        <field name="name">Allocations: base.group_user create/write</field>
         <field name="model_id" ref="model_hr_leave_allocation"/>
-        <field name="domain_force">
-            [('holiday_type', '=', 'employee'),
+        <field name="domain_force">[
+            ('holiday_status_id.allocation_type', '=', 'fixed_allocation'),
+            ('holiday_type', '=', 'employee'),
             '|',
                 '&amp;',
                     ('employee_id.user_id', '=', user.id),
                     ('state', 'not in', ['validate', 'validate1']),
                 '&amp;',
-                    ('validation_type', 'in', ['manager', 'both', 'no_validation']),
-                    ('employee_id.leave_manager_id', '=', user.id),]
-        </field>
+                    ('validation_type', 'in', ['manager', 'both']),
+                    ('employee_id.leave_manager_id', '=', user.id),
+        ]</field>
         <field name="perm_read" eval="False"/>
+        <field name="perm_unlink" eval="False"/>
         <field name="groups" eval="[(4,ref('base.group_user'))]"/>
     </record>
 
+    <record id="hr_leav_allocation_rule_employee_unlink" model="ir.rule">
+        <field name="name">Allocations base.group_user unlink</field>
+        <field name="model_id" ref="model_hr_leave_allocation"/>
+        <field name="domain_force">[('employee_id.user_id', '=', user.id), ('state', '=', 'draft')]</field>
+        <field name="perm_read" eval="False"/>
+        <field name="perm_write" eval="False"/>
+        <field name="perm_create" eval="False"/>
+        <field name="groups" eval="[(4, ref('base.group_user'))]"/>
+    </record>
+
     <record id="hr_leave_allocation_rule_officer_read" model="ir.rule">
```

Comment:
With https://github.com/odoo/odoo/pull/35446/files#diff-fcf8db67e1f86a7cf96fc12fc27eb9f3R96, this rule is useless now.